packages/core/src/Cardano/types/Key.ts

```diff
@@ -46,18 +47,22 @@ Ed25519PrivateKey.fromHexBlob = (value: HexBlob) => castHexBlob<Ed25519PrivateKe
 export type Ed25519KeyHash = OpaqueString<'Ed25519KeyHash'>;
 export const Ed25519KeyHash = (value: string): Ed25519KeyHash => Hash28ByteBase16(value);
 Ed25519KeyHash.fromRewardAccount = (rewardAccount: RewardAccount): Ed25519KeyHash =>
-  Ed25519KeyHash(
-    Buffer.from(
-      CSL.RewardAddress.from_address(CSL.Address.from_bech32(rewardAccount.toString()))!
-        .payment_cred()!
-        .to_keyhash()!
-        .to_bytes()
-    ).toString('hex')
+  usingAutoFree((scope) =>
+    Ed25519KeyHash(
+      Buffer.from(
+        scope
+          .manage(CSL.RewardAddress.from_address(scope.manage(CSL.Address.from_bech32(rewardAccount.toString())))!)
+          .payment_cred()!
+          .to_keyhash()!
+          .to_bytes()
+      ).toString('hex')
+    )
   );
 
-Ed25519KeyHash.fromKey = (pubKey: Ed25519PublicKey): Ed25519KeyHash => {
-  const cslPubKey = CSL.PublicKey.from_bytes(Buffer.from(pubKey, 'hex'));
-  const keyHash = cslPubKey.hash().to_bytes();
+Ed25519KeyHash.fromKey = (pubKey: Ed25519PublicKey): Ed25519KeyHash =>
+  usingAutoFree((scope) => {
+    const cslPubKey = scope.manage(CSL.PublicKey.from_bytes(Buffer.from(pubKey, 'hex')));
+    const keyHash = cslPubKey.hash().to_bytes();
```

Comment:
hash() probably needs to be managed

Reply:
214d62f11644c689a1e38c3f1811f88e87acf8ac